panther_lights/src/driver_node.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+from dataclasses import dataclass
+from threading import Lock
+
+from apa102_pi.driver import apa102
+import RPi.GPIO as GPIO
+
+import rospy
+
+from sensor_msgs.msg import Image
+
+from panther_msgs.srv import SetLEDBrightness, SetLEDBrightnessRequest, SetLEDBrightnessResponse
+
+
+@dataclass
+class LEDConstants:
+    LED_SWITCH_FRONT_STATE = GPIO.HIGH
+    LED_SWITCH_REAR_STATE = GPIO.LOW
+    LED_POWER_ON_STATE = GPIO.LOW  # active LED with low state
+    LED_SWITCH_PIN = 20
+    LED_POWER_PIN = 26
+    LED_MAX_BRIGHTNESS = 31
+    PANEL_FRONT = 0
+    PANEL_REAR = 1
+
+class LightsDriverNode:
+    def __init__(self, name: str) -> None:
+        rospy.init_node(name, anonymous=False)
+
+        global_brightness = rospy.get_param('~global_brightness', 1.0)
+        self._frame_timeout = rospy.get_param('~frame_timeout', 0.1)
+        self._num_led = rospy.get_param('~num_led', 46)  # has to match controller num_led
+
+        try:
+            apa_driver_brightness = self._percent_to_apa_driver_brightness(global_brightness)
+        except ValueError as err:
+            rospy.logwarn(f'[{rospy.get_name()}] Invalid brightness: {err}. Using default')
+            apa_driver_brightness = LEDConstants.LED_MAX_BRIGHTNESS
+
+        self._lock = Lock()
+        self._pixels = apa102.APA102(
+            num_led=self._num_led,
+            order='rgb',
+            mosi=10,
+            sclk=11,
+            global_brightness=apa_driver_brightness,
+        )
+        self._front_active = True
+        self._rear_active = True
+        color_correction_rgb = [255, 200, 62]
+        # rgb values normalized to avoid additional division
+        self._color_correction = [value / 255 for value in color_correction_rgb]
+
+        # setup and activate panels
+        self._setup_panels()
+
+        # clear panels
+        self._clear_panel(LEDConstants.PANEL_FRONT)
+        self._clear_panel(LEDConstants.PANEL_REAR)
+
+        # -------------------------------
+        #   Subscribers
+        # -------------------------------
+
+        self._front_frame_sub = rospy.Subscriber(
+            'lights/driver/front_panel_frame', Image, self._front_frame_cb, queue_size=1
+        )
+        self._rear_frame_sub = rospy.Subscriber(
+            'lights/driver/rear_panel_frame', Image, self._rear_frame_cb, queue_size=1
```

Comment:
```suggestion
        self._front_frame_sub = rospy.Subscriber(
            'lights/driver/front_panel_frame', Image, self._front_frame_cb, queue_size=3
        )
        self._rear_frame_sub = rospy.Subscriber(
            'lights/driver/rear_panel_frame', Image, self._rear_frame_cb, queue_size=3
```

I would extend queue size and check if timestamp of new message is not from past